sqlobject/src/main/java/org/jdbi/v3/sqlobject/SqlObjectFactory.java

```diff
@@ -127,6 +128,17 @@ private boolean looksLikeSqlObject(Class<?> extensionType) {
                 handlers.computeIfAbsent(method, m -> buildMethodHandler(sqlObjectType, m, registry, decorators));
             }
 
+            handlers.keySet().stream()
+                .filter(m -> !m.isSynthetic())
+                .collect(Collectors.groupingBy(m -> Arrays.asList(m.getName(), Arrays.asList(m.getParameterTypes()))))
+                .values()
+                .stream()
+                .filter(l -> l.size() > 1)
+                .findAny()
+                .ifPresent(ms -> {
+                    throw new UnableToCreateSqlObjectException(sqlObjectType + " has ambiguous methods " + ms);
```

Comment:
This error message would be more helpful if it included a suggestion to override the method and annotate it / provide a `default` implementaiton in `sqlObjectType`